assets/src/hooks/useLadderSvgHeight.ts

```diff
@@ -0,0 +1,35 @@
+import { Dispatch, RefObject, SetStateAction, useLayoutEffect } from "react"
+
+const useLadderSvgHeight = (
+  ref: RefObject<SVGSVGElement>,
+  x: number,
+  y: number,
+  width: number,
+  height: number,
+  setHeight: Dispatch<SetStateAction<number>>
+) => {
+  useLayoutEffect(() => {
+    if (
+      ref.current !== null &&
+      // only if there is no viewBox attribute
+      !ref.current.getAttribute("viewBox") &&
+      // only if rendered
+      ref.current.getBBox().width &&
+      ref.current.getBBox().height
+    ) {
+      const routeLadder: HTMLElement = document.getElementsByClassName(
+        "m-ladder"
+      )[0] as HTMLElement
```

Comment:
Should this element be gotten by `useRef` instead of `document.getElement`?

All of the heights are set based on the height of the first ladder. How does that compare to each ladder checking itself for the height.

Reply:
No, the svg doesn't have the full height yet, we need to get the height of the bounding box in order to tell the svg to fill it. It doesn't matter that we use the first ladder div because they are all the same height.